pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -248,10 +256,11 @@ def _map_message(cls, message: ModelMessage) -> Iterable[chat.ChatCompletionMess
         else:
             assert_never(message)
 
-    @classmethod
-    def _map_user_message(cls, message: ModelRequest) -> Iterable[chat.ChatCompletionMessageParam]:
+    def _map_user_message(self, message: ModelRequest) -> Iterable[chat.ChatCompletionMessageParam]:
         for part in message.parts:
             if isinstance(part, SystemPromptPart):
+                if self.system_prompt_role == 'developer':
+                    yield chat.ChatCompletionDeveloperMessageParam(role='developer', content=part.content)
                 yield chat.ChatCompletionSystemMessageParam(role='system', content=part.content)
```

Comment:
```suggestion
                if self.system_prompt_role == 'developer':
                    yield chat.ChatCompletionDeveloperMessageParam(role='developer', content=part.content)
                else:
                    yield chat.ChatCompletionSystemMessageParam(role='system', content=part.content)
```
Otherwise it will always yield the system prompt after yielding the developer prompt

Reply:
Ah yes, I had my `return` headset on.